### DISCUSSION TITLE
Is it possible to style the CodeMirror editor in My Rules (in Chromium)?

### DISCUSSION THREAD

**Discussion by tomasz1986** [uBlockOrigin/uBlock-issues#2430] [Q&A] [answered]
This is possible in Firefox through the built-in CSS customisation, but Chromium doesn't allow to apply custom CSS to its internal pages which include extension-specific pages.

Specifically, I would like to increase the font size and line height in the rules list, as I've got poor eyesight, and the defaults are difficult to read, especially when there are many lines there. For now I've only managed to increase the font size by changing the `--monospace-size` variable in https://github.com/gorhill/uBlock/wiki/Advanced-settings#uistyles, but there doesn't seem to be any similar variable responsible for line height.

When it comes to the line height, it can be controlled by applying custom `line-height` to the `.CodeMirror-line` class, but as explained above, this can't be done in Chromium, and https://github.com/gorhill/uBlock/wiki/Advanced-settings#uistyles can only control `body` styles (which in this case have no effect on the line height used in the CodeMirror editor).

**Comment by gorhill** [comment]
You can use page zoom on extension pages.

**Reply by tomasz1986** [reply]
That's an option, of course, although it's not much different than increasing `--monospace-size` which I already do 😉. Being able to increase the line height would still be helpful though. In fact, `body` already uses a nice line height of `1.5`, however CodeMirror overwrites it with its own `1.25` 🙁.

**Reply by gorhill** [reply]
Well the difference is that I have nothing to do, and relying on CodeMirror's internal DOM structure is not a good idea anyway -- eventually I will move to CodeMirror 6 and all your customizations will probably be broken.

Anyway, that kind of customization requests is not something I will spent time on, I rather spent time I set aside for this project on stuff which is meaningful for a majority of users, or to make the life easier for filter list maintainers who contribute to the project.

**Reply by tomasz1986** [reply]
Understood. This was more of a question than a proper feature request. I was curious whether there was maybe another way/hack to apply custom CSS to the thing. I'll probably just live with the current state since with at least the larger font, the text is readable enough.

**Edit**: To be honest, personally I'm not a fan of CodeMirror (or other similar code editors for that matter). They're usually poor in terms of accessibility and require heavy tweaking to make them pleasant to use 🙁.

**Comment by tomasz1986** [comment, marked as answer]
Just for information, I've managed to find a workaround. It is a bit clunky, but it does work.

1. Open the My Rules page, then open DevTools (F12).
2. Add the custom CSS to local overrides (see https://developer.chrome.com/blog/new-in-devtools-65/#overrides).

The clunky part is that the styling will persist only a) while DevTools is open or b) until the page is reloaded if DevTools has been closed. This means that you must open DevTools at least once upon opening the uBlock internal page to apply the style for the first time.

Despite the above, I will still probably find this useful when having to stay on the page for a while in order to edit many rules.
